Source/SvgDocument.cs

```diff
@@ -280,181 +342,143 @@ public static T Open<T>(Stream stream, Dictionary<string, string> entities) wher
 
         private static T Open<T>(XmlReader reader) where T : SvgDocument, new()
         {
-            try
-            {
-                var elementStack = new Stack<SvgElement>();
-                bool elementEmpty;
-                SvgElement element = null;
-                SvgElement parent;
-                T svgDocument = null;
-                var elementFactory = new SvgElementFactory();
+            EnsureSystemIsGdiPlusCapable(); //Validate whether the GDI+ can be loaded, this will yield an exception if not
+            var elementStack = new Stack<SvgElement>();
+            bool elementEmpty;
+            SvgElement element = null;
+            SvgElement parent;
+            T svgDocument = null;
+            var elementFactory = new SvgElementFactory();
 
-                var styles = new List<ISvgNode>();
+            var styles = new List<ISvgNode>();
 
-                while (reader.Read())
+            while (reader.Read())

```

Comment:
This is now outside the exception handler. I this intentional?

Reply:
I checked the older versions, and it seems that prior my changes this code was outside of the try catch. During the initial changes I was not sure where the issues regarding GDI+ were raised, so I put the whole call chain in a try/catch, with additions to the original catch blocks to validate the exception type. Since we now perform the Ensure function at the start of the call we no longer need the try catch construction regarding the GDI+ errors. Other errors are not to be expected, so I thinks it's safe to leave it out, the errors will be raised to the caller since I am not sure how and what we should handle if we put this in a big try/catch.